### PR TITLE
Improve namespacing for chapter 1

### PR DESCRIPTION
**Pull request by MAO3J1m0Op**
Because many of our names conflict with those defined in Mathlib, the current paradigm of namespacing and importing used in chapter 1 will not work when future chapters are introduced, as using the `open` keyword creates definition ambiguity between our definition and Mathlib's definition that Lean cannot solve.

Instead, we will place everything in one namespace. However, our currently used name of `Defs` is not very descriptive, so we'll instead use the namespace name `AlgebraInLean`.

### CODE REVIEW

**Review by MithicSpirit — CHANGES_REQUESTED** [review]
Imo this should delete Basic.lean as well.

**Review by Stavan-Jain — APPROVED** [review]
lgtm

File: AlgebraInLean.lean
Comment (MithicSpirit) [inline AlgebraInLean.lean:1]: Maybe run `lake exe mk_all` to build this, although that's not really necessary for this PR atm.

**Review by MithicSpirit — APPROVED** [review]
lgtm

**Comment by justin-t-morrill** [comment]
lgtm

**Review by justin-t-morrill — APPROVED** [review]

**Review by AdamJasonKern — APPROVED** [review]

**Pull request merged**